### DISCUSSION TITLE
[Help Request] IDE Handling

### DISCUSSION THREAD

**Discussion by GregOnNet** [nx-dotnet/nx-dotnet#84] [Q&A]
Hi, 

thank you for integrating .NET with NX.
Currently, I am evaluating this plugin for our project.

## Seen Behaviour
Generating Apps and Libs works as expected.
I am also able to link a library to an application, modifying the .csproj-File.

Unfortunately, Rider does not recognize generated libraries as .NET projects.

## Question
Can you provide some guidance how the correct setup should look like?

![CleanShot 2021-07-16 at 16 52 13@2x](https://user-images.githubusercontent.com/444278/125967169-7f0229bb-bf4b-434a-8ebe-525a26092012.png)

## Current Workaround

I added the solution file manually to make Rider aware of generated projects.

```bash
# Here is what I did
dotent new sln # generates nx-dotnet-playground.sln
dotnet sln add apps/api-functions
dotnet sln add apps/api-functions-test
dotnet sln add libs/mylib

# close Rider
# open Rider by double-clicking nx-dotnet-playground.sln in Explorer/Finder.
```

## Demo Repository
You can see my work at: https://github.com/GregOnNet/nx-dotnet-playground

Thanks in advance for your help.

**Comment by AgentEnder** [comment]
Hey @GregOnNet. I've been meaning to add something like "@nx-dotnet/core:solution" as an executor, that would generate a temporary solution file for using in IDE's. As is, development in vs-code goes pretty smoothly since it does not care about / use sln files, but I know that is not necessarily ideal for .NET.

 Most of the work is already there, I've just been going between a generator vs executor internally for a while now.

Generator makes sense, as it generates a file. On the other hand, its a temporary file that could change as project refs are added.... That makes me think an executor. I'll get a rough version of this out sometime before monday, so it can be played around with at least.

**Comment by GregOnNet** [comment]
Thanks for sharing your thoughts.

By temporary file you mean it is highly dynamic, right?

My initial thought is
- adding a flag to the generate app-Command: sln true
- If no sln exists a new one is created
- After this each generate library or generate app looks for the sln-File... If sln is present the new app or library will be added to it

Another idea
- providing a flag to generate library that allows to link the library to another project

These are just a few thoughts...

**Comment by AgentEnder** [comment]
Hey, sorry. Really expected this to be done before vacation 🙍‍♂️. I'll try to swing back to this sometime this week.

**Comment by GregOnNet** [comment]
Hey @AgentEnder,

no worries! Using the dotnet CLI creating the Solution File totally works for us.
